generate_metap_agg.py

```diff
@@ -1,193 +1,465 @@
-import os
 import requests
-from pymongo import MongoClient
-
-
-DEBUG_MODE = False
+import csv
+import io
+import os
+import time
+from abc import ABC, abstractmethod
 
+# TODO KRH: Change metagmetagenome_anlaysis_id to was_generated_by throughout after https://github.com/microbiomedata/nmdc-schema/pull/2203 has been merged and data have been migrated
 
-class AnnotationLine():
+class Aggregator(ABC):
     """
-    Object representation built from a GFF annotation line
+    Abstract class for Aggregators
 
-    The object countains a list of annotations for KEGG,
-    COG, EC, and product.  Currently on the KEGG annotations
-    are used.
+    Parameters
+    ----------
+    dev : bool
+        Flag to indicate if production or development API should be used
+        Default is True, which uses the development API
+    
+    Attributes
+    ----------
+    base_url : str
+        Base URL for the API, either production or development
+        "https://api.microbiomedata.org" or "https://api-dev.microbiomedata.org"
+    nmdc_api_token : str
+        API bearer token to access the API
+    aggregation_filter : str
+        Filter to apply to the aggregation collection endpoint to get applicable records (set in subclasses)
+        e.g. '{"metagenome_annotation_id":{"$regex":"wfmp"}}'
+    workflow_filter : str
+        Filter to apply to the workflow collection endpoint to get applicable records (set in subclasses)
+        e.g. '{"type":"nmdc:MetaproteomicsAnalysis"}'
     """
+    def __init__(self, dev=True):
+        if dev:
+            self.base_url = "https://api-dev.microbiomedata.org"
```

Comment:
This could be an environmental variable as well.

Reply:
you'd set the values for these permanently in SPIN. So the namespace nmdc-dev in SPIN, which has an aggregation deployment,  would have a variable name NMDC_RUNTIME_URL with a value of https://api-dev.microbiomedata.org and the namespace nmdc (this is what prod is called) in SPIN, which has an aggregation deployment, would have a variable name NMDC_RUNTIME_URL with a value of https://api.microbiomedata.org

Reply:
Refactored to set the base_url attribute from an (optional) environmental variable in script and in readme.